packages/flutter/lib/src/widgets/sheet.dart

```diff
@@ -0,0 +1,1036 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'layout_builder.dart';
+import 'media_query.dart';
+import 'primary_scroll_controller.dart';
+import 'scroll_context.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+import 'scroll_position_with_single_context.dart';
+import 'scroll_simulation.dart';
+import 'ticker_provider.dart';
+import 'transitions.dart';
+
+// TODO(jamesblasco): Arbitrary values
+const double _kWillPopThreshold = 0.8;
+const Curve _kDecelerateEasing = Cubic(0.0, 0.0, 0.2, 1.0);
+const Duration _kDefaultSheetDuration = Duration(milliseconds: 400);
+const Duration _kBounceAnimationDuration = Duration(milliseconds: 300);
+
+/// NEEDS DOCS
```

Comment:
minor hint: if you just skip the docs entirely, the linter will tell you where docs are missing (but when you put placeholder docs with `///`, it won't)